gradle/validation/error-prone.gradle

```diff
@@ -0,0 +1,148 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+allprojects { prj ->
+  plugins.withType(JavaPlugin) {
+    prj.apply plugin: 'net.ltgt.errorprone'
+
+    dependencies {
+      errorprone("com.google.errorprone:error_prone_core")
+    }
+
+    tasks.withType(JavaCompile) { task ->
+      options.errorprone.errorproneArgs = [
+          // test
+          '-Xep:ExtendingJUnitAssert:OFF',
```

Comment:
Personal style, but I think ```options.errorprone {
disable 'ExtendingJUnitAssert'
}```
is more clear than using `-Xep`?

Reply:
I am okay with either of the two styles. Ideally we'd want this list to get much shorter soon :)

Reply:
Varun - feel free to take this branch (or patch) and roll it out on yours. I didn't intend it to be committed, I just wanted to show what's needed for it to compile and work.

Reply:
sounds good! I'll take the current branch and add two things and then commit the code
1. `options.errorprone.disableWarningsInGeneratedCode = true`
2. CHANGES entry